### PR TITLE
vprobe electrodes locations

### PR DESCRIPTION
**Pull request by luiztauffer**
This change reads the relative-to-probe electrodes locations and add it to the recording. This is important for spike sorting

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
Looks like you're setting it all explicitly

Can you instead make a ProbeInterface.Probe object and use `recording_interface.set_probe` instead?

**Comment by luiztauffer** [comment]
@CodyCBakerPhD what would be the advantage of doing that?

**Comment by CodyCBakerPhD** [comment]
It's a more standard workflow, is a more consolidatted injection, results in more standard electrode table structure (since it goes through a specific pathway for probes in the write tools)

**Comment by luiztauffer** [comment]
@CodyCBakerPhD sounds good to me! Could you point me to an example converter doing this?

**Comment by CodyCBakerPhD** [comment]
Example in tests: https://github.com/catalystneuro/neuroconv/blob/9a72cdf9f341359bb4058fa91260dcf595d800c1/src/neuroconv/tools/testing/data_interface_mixins.py#L483

Example for SpikeGLX (probe loaded via helpers): https://github.com/catalystneuro/neuroconv/blob/9a72cdf9f341359bb4058fa91260dcf595d800c1/src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py#L89

Tutorial for created a custom probe: https://probeinterface.readthedocs.io/en/main/examples/ex_01_generate_probe_from_sratch.html

**Comment by alejoe91** [comment]
@luiztauffer @CodyCBakerPhD 

I implemented the `set_probe` and also changed the interface to use the `BinaryRecording`, so we don't need to duplicate all the memmapping logic. I think the result is much cleaner and simple. @luiztauffer can you test it out? I don't have the raw data on my side

File: src/jazayeri_lab_to_nwb/watters/wattersrecordinginterface.py [hunk 1]
```diff
@@ -89,36 +28,55 @@ def __init__(
...
+        probe.name = probe_name
+
+        # set probe to interface recording
+        self.set_probe(probe)
+
+        # set group_name property to match electrode group name in metadata
+        self.recording_extractor.set_property(
+            key="group_name",
+            values=[probe_name] * len(self.recording_extractor.channel_ids),
+        )
```
Comment (CodyCBakerPhD) [inline src/jazayeri_lab_to_nwb/watters/wattersrecordinginterface.py:80]: Doesn't Jeremy's function do this bit already? https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py#L249

**Comment by luiztauffer** [comment]
@alejoe91 I'm getting this error with the `BinaryRecording`

```
File /mnt/shared_storage/Github/neuroconv/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py:32, in BaseRecordingExtractorInterface.__init__(self, verbose, es_key, **source_data)
     20 """
     21 Parameters
     22 ----------
   (...)
     29 
     30 """
     31 super().__init__(**source_data)
---> 32 self.recording_extractor = self.get_extractor()(**source_data)
     33 self.subset_channels = None
     34 self.verbose = verbose

File /mnt/shared_storage/Github/neuroconv/src/neuroconv/baseextractorinterface.py:25, in BaseExtractorInterface.get_extractor(cls)
     23     return cls.Extractor
     24 extractor_module = get_package(package_name=cls.ExtractorModuleName)
---> 25 extractor = getattr(
     26     extractor_module,
     27     cls.ExtractorName or cls.__name__.replace("Interface", "Extractor"),
     28 )
     29 cls.Extractor = extractor
     30 return extractor

AttributeError: module 'spikeinterface.extractors' has no attribute 'BinaryRecording'
```

----------------------------------------------------------------

**Edit 1**: It works with `BinaryRecordingExtractor`, but other errors show up, I'll keep debugging
```
  File "/mnt/shared_storage/Github/jazayeri-lab-to-nwb/src/jazayeri_lab_to_nwb/watters/wattersrecordinginterface.py", line 47, in __init__
    super().__init__(verbose=verbose, es_key=es_key, **source_data)
  File "/mnt/shared_storage/Github/neuroconv/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py", line 32, in __init__
    self.recording_extractor = self.get_extractor()(**source_data)
  File "/home/luiz/anaconda3/envs/env_jaz/lib/python3.10/site-packages/spikeinterface/core/binaryrecordingextractor.py", line 113, in __init__
    self.set_channel_gains(gain_to_uV)
  File "/home/luiz/anaconda3/envs/env_jaz/lib/python3.10/site-packages/spikeinterface/core/baserecordingsnippets.py", line 381, in set_channel_gains
    self.set_property("gain_to_uV", gains, ids=channel_ids)
  File "/home/luiz/anaconda3/envs/env_jaz/lib/python3.10/site-packages/spikeinterface/core/base.py", line 194, in set_property
    assert values.shape[0] == size
AssertionError
```

**Comment by alejoe91** [comment]
@luiztauffer should be fixed now :) gains and offsets needed to be a scalar rather than an array.

@CodyCBakerPhD you need to set group_name because the `get_metadata` sets a different custom group name, so we have to make sure that it corresponds to the `group_name` property!

**Comment by CodyCBakerPhD** [comment]
@luiztauffer Any updates on this PR? Is it producing  good files and do those files work with dendro?

**Comment by luiztauffer** [comment]
@CodyCBakerPhD the conversion for Vprobe is working now, electrodes locations are correct

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**